spock-core/src/main/java/org/spockframework/mock/EmptyOrDummyResponse.java

```diff
@@ -59,6 +74,10 @@ public Object respond(IMockInvocation invocation) {
       return ReflectionUtil.getDefaultValue(returnType);
     }
 
+    if (returnType.isAssignableFrom(invocation.getMockObject().getType())) {
+      return invocation.getMockObject().getInstance();
```

Comment:
Yes it does, however this is just the default answer, you can still define your answer if you really care. Before you'd get a normal instance of `Object` and now you get the stub object I don't think that this changes much.

This was possible before in Spock `Service stub = Stub {_(*_) >> stub}` although this would match every other call. I'm thinking of replacing the default `ZeroOrNullResponse` with `ZeroOrNullOrSelfResponse` for `Mock`s so that they also support the builder pattern.